gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -325,7 +327,7 @@ public EntityResponse visit(SelectionNode selectionNode) {
                                   .getExpressionContext()
                                   .getSourceToTimeAggregationMap()
                                   .get(source))
-                          .setFilter(filter)
+                          .setFilter(addSourceFilters(executionContext, source, filter))
```

Comment:
What about total? Should we send the same filter for the total request?

Reply:
total is fetched as part of data fetcher node, so all filters will be used